config/dynamic.c

```diff
@@ -291,6 +292,175 @@ cleanup_dy(struct dynamic_config *dy_conf)
 		rte_free(dy_conf->server_path);
 		dy_conf->server_path = NULL;
 	}
+
+	destroy_mailbox(&dy_conf->mb);
+}
+
+static void
+process_return_message(lua_State *l, struct dynamic_config *dy_conf,
+	int num_succ_sent_inst)
+{
+	int num_gt_messages = 0;
+	size_t reply_len = 0;
+	char reply_msg[MSG_MAX_LEN];
+	int num_cmd_acc = 0;
+
+	/* Wait for all GT instances to synchronize. */
+	while (rte_atomic16_read(&dy_conf->num_returned_instances)
+			< num_succ_sent_inst)
+		rte_pause();
+
+	while (num_cmd_acc < num_succ_sent_inst) {
+		int i;
+		struct dy_cmd_entry *dy_cmds[dy_conf->mailbox_burst_size];
+		/* Load a set of commands from its mailbox ring. */
+		int num_cmd = mb_dequeue_burst(&dy_conf->mb,
+			(void **)dy_cmds, dy_conf->mailbox_burst_size);
+
+		for (i = 0; i < num_cmd; i++) {
+			struct dy_cmd_entry *entry = dy_cmds[i];
+			switch (entry->op) {
+				case GT_UPDATE_POLICY_RETURN: {
+
+					if (dy_conf->gt == NULL) {
+						DYC_LOG(ERR, "The command operation %u requires that the server runs as Grantor\n",
+							entry->op);
+						break;
+					}
+
+					if (unlikely(entry->u.gt.length > MSG_MAX_LEN))
+						DYC_LOG(ERR, "The return message from GT block is too long\n");
+					else if (reply_len + entry->u.gt.length >=
+							MSG_MAX_LEN)
```

Comment:
1. Since the string is binary now, we can replace `>=` with `>`.

2. Use `unlikely()` here too.